numpy/core/shape_base.py

```diff
@@ -177,6 +177,71 @@ def atleast_3d(*arys):
         return res
 
 
+def atleast_nd(n, *arys):
+    """
+    View inputs as arrays with at least `n` dimensions.
+
+    Parameters
+    ----------
+    n : scalar
+        The minimum number of dimensions required for each of the input
+        arrays.
+    arys1, arys2, ... : array_like
+        One or more array-like sequences. Non-array inputs are
+        converted to arrays.  Arrays that already have `n` or more
+        dimensions are preserved.
+
+    Returns
+    -------
+    res1, res2, ... : ndarray
+        An array, or tuple of arrays, each with ``a.ndim >= n``. Copies
+        are avoided where possible, and views with `n` or more
+        dimensions are returned. Dimensions are always prepended, so for
+        example, a 1-D array of shape ``(N,)`` with at least 4
+        dimensions required becomes a view of shape ``(1, 1, 1, N)``,
+        and a 2-D array of shape ``(M, N)`` becomes a view of shape
+        ``(1, 1, M, N)``.
+
+    See Also
+    --------
+    atleast_1d, atleast_2d, atleast_3d
+
+    Examples
+    --------
+    >>> np.atleast_nd(4, 3.0)
+    array([[[[ 3.]]]])
+
+    >>> x = np.arange(3.0)
+    >>> np.atleast_nd(2, x).shape
+    (1, 3)
+
+    >>> x = np.arange(12.0).reshape(4, 3)
+    >>> np.atleast_nd(5, x).shape
+    (1, 1, 1, 4, 3)
+    >>> np.atleast_nd(5, x).base is x
+    True
+
+    >>> for arr in np.atleast_nd(2, (1, 2), [[1, 2]], [[[1, 2]]]):
+    ...     print(arr, arr.shape)
+    ...
+    [[1 2]] (1, 2)
+    [[1 2]] (1, 2)
+    [[[1 2]]] (1, 1, 2)
+
+    """
+    res = []
+    for ary in arys:
+        ary = asanyarray(ary)
```

Comment:
I did not use your suggestion literally because zero-dim arrays need special handling, but I did something quite similar, just without the `ndmin` parameter (it's not `ndim` for `array`).